src/Common/RipeMD160Hash.h

```diff
@@ -0,0 +1,191 @@
+#pragma once
+
+#include <bit>
+#include <string>
+#include <type_traits>
+#include <Core/Defines.h>
+#include <base/extended_types.h>
+#include <base/hex.h>
+#include <base/types.h>
+#include <base/unaligned.h>
+#include <boost/multiprecision/cpp_int.hpp>
+#include <Common/Exception.h>
+#include <Common/transformEndianness.h>
+
+#include <city.h>
+
+/// https://homes.esat.kuleuven.be/~bosselae/ripemd160/pdf/AB-9601/AB-9601.pdf
+/// https://en.wikipedia.org/wiki/RIPEMD
+
+class RipeMD160
```

Comment:
Isn't it available in OpenSSL?

Reply:
yep! I think it's reasonable to use OpenSSL version, cause my implementation is a bit costy. made changes and added info in docs. thank you!